device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -288,10 +288,11 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
     std::map<unsigned int, vecmem::data::vector_buffer<candidate_link>>
         link_map;
 
-    // Create a map for tip links
-    std::map<unsigned int, vecmem::data::vector_buffer<
-                               typename candidate_link::link_index_type>>
-        tips_map;
+    // Create a buffer of tip links
+    vecmem::data::vector_buffer<typename candidate_link::link_index_type>
+        tips_buffer{m_cfg.max_num_branches_per_seed * n_seeds, m_mr.main,
```

Comment:
🤔 How expensive is this over-allocation? Since with this we allocate a little more than with the current code, right?

Though I guess `candidate_link::link_index_type` is a 32-bit integer anyway, right? So this should not be too worrisome either way I guess. 🤔

Reply:
`max_num_branches_per_seed` is (currently) 10, `n_seeds` is probably somewhere around 50,000 for ⟨μ⟩, so we're talking about around 2 MB. Pretty negligible, I'd say.